### PR TITLE
chore(content): add hidden debug page

### PR DESCRIPTION
**Pull request by k88hudson**
I was thinking this might be useful for debugging -- it adds a little link in the bottom right corner if you hover over it for 0.2s.

What do you think @nchapman @pdehaan?

### CODE REVIEW

**Comment by pdehaan** [comment]
I think it's definitely useful, but not sure if its too easily discoverable (unless this is a temporary hack and we'll pull it out before release). What about hiding it behind an `about:config` pref?

Also, given the amount of data being returned by our Embedly proxy, not sure if it'd be useful to use something like https://www.npmjs.com/package/react-json-tree to easily debug/collapse the JSON tree. But then again, probably not worth bloating up the vendor.bundle.js.

**Comment by pdehaan** [comment]
That said, this was super useful in trying to debug why I wasn't getting the correct data for a bugzilla URL.
Not sure why I'm not seeing any Embedly data injected into that record (but I do see it for the npmjs.com result):

```json
{
  "TopSites": {
    "rows": [
      {
        "url": "https://bugzilla.mozilla.org/show_bug.cgi?id=1255136",
        "title": "1255136 – Deploy TestPilot 2016-03-09",
        "lastVisitDate": 1457558902213,
        "frecency": 2000,
        "favicon": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAC2klEQVQ4jYWSbyzUcRzHvxeai8r050GPcMIdd3H3c/fzu8M5fzZdx4XrGGKWIazj/C0zI6Qdskg1Tofz967Lv9QWSmvZanrSVls9aOtJra1FKSa9e1Qo6f3svX0+r73f25uQbRQ6ySB0RBpGCCHSYTpOZqGWt7snitsizkYfMkpDZgkcy6lXetD9wkzGLPy8LUA+yUA+Kin65ZnhwBXpgPjjqcqIvpg6ZpnpEEFbEHTxnwDZiGRBahOvEEKIbJheZAYCEWSmEGSiUDlbiNm3NpQZ0nA0w794685WSZhsWAxZv2RBaYmE9bkRd170oulhLfjNPkibjMXQXDsCGv3g3+ALQS0PJNhGL4ps690ZM4VQcxAG5i/h/N0MnBtNRMloOkLaafAbuMjo1UKhPvx6PbZVsiodEoPpo3pCesQfTljVMD9tRNXESeitsUjtUSBp6DgGn1xH20wTBDVcMFrOrU3RJd2iOdokAm0UwfKsFZXjKSi0xCDVFAZtXwyuPqiFfkCLa9N14FfywCv3XPIp5uRvgog7hCtJg3Ew3MuHbkiFlC4FErqP4cpMDXRmDVILIqHLjIX6rBKBZQJ46zjwzuNgPcVl4ddsawZapkuReiMC8V1KGMbLkNuZhCjGFzWObEy52KHIdT/8/APeiAAHTpZ7xW8A3Sy6IDTwUT1RigSjCoaxcpw2qeCnpBDvuBPv7Xeg3YWFfmenNfPefTA6sx/r2tr8N9UQ1gpmOUVu6LzfimpbHrKMKshiFJg46IxHLIJ3hIVBQqDfvWcVu9gwuLh++WsL/HLurKCchzOmbOi7c+ERdQQlDmzME/JjisXCTUKQ43oAFWGhyPJzX9pyUFy9V7dXvvuaZ447uNEUkr098YmQ7y8JWQtxO7TYrE6sS+fx6tXtVfItARtF5VPzyRolXhGCXlc2ZFHhLf99+lMead6ridFe31T2Ts3hKenBhBCSLpc7ajQau5+bn1TH+A/7owAAAABJRU5ErkJggg==",
        "type": "history",
        "parsedUrl": {
          "hash": "",
          "query": {},
          "protocol": "https:",
          "pathname": "/show_bug.cgi",
          "auth": "",
          "host": "bugzilla.mozilla.org",
          "port": "",
          "hostname": "bugzilla.mozilla.org",
          "password": "",
          "username": "",
          "href": "https://bugzilla.mozilla.org/show_bug.cgi"
        }
      },
      {
        "provider_url": "https://www.npmjs.com",
        "authors": [],
        "provider_display": "www.npmjs.com",
        "related": [],
        "favicon_url": "https://www.npmjs.com/static/images/touch-icons/favicon.ico",
```

Weirder still, it looks like Embedly proxy is returning the data, we're just not handling it (i'm guessing because the original URL had a `?id=xxx` but the sanitized link we pass to embedly doesn't):
new_tab_and_http___embedly-proxy_dev_mozaws_net_extract_urls_https___bugzilla_mozilla_org_show_bug_cgi

**Comment by pdehaan** [comment]
R+.
Was useful. Would debug again.

**Comment by k88hudson** [comment]
@pdehaan I added in an inspector -- luckily due to our DefinePlugin, anything required inside a `if(__CONFIG__.DEVELOPMENT)` block gets omitted for the production build :dancer:

**Comment by k88hudson** [comment]
ah crap hang on lint issues with the vendor css

File: content-src/components/NewTabPage/NewTabPage.scss [hunk 1]
```diff
@@ -3,3 +3,20 @@
...
+  bottom: 0;
+  right: 0;
+  padding: 10px;
+  background: $link-blue;
+  color: $white;
+  opacity: 0;
+  z-index: 1000;
+  font-size: 10px;
+  transition: 0.2s 0.2s opacity ease;
+  &:hover {
```
Comment (pdehaan) [inline content-src/components/NewTabPage/NewTabPage.scss]: nit: add a blank row before the `&:hover`:

```
18:1  warning  Space expected between blocks  empty-line-between-blocks
```

File: content-src/components/DebugPage/DebugPage.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          {JSON.stringify(this.props.newTab, null, 2)}
+        </pre>
+      </div>
+    </main>);
+  }
+});
+
+module.exports = connect(state => {
+  return {
+    newTab: dedupedSites(state),
```
Comment (pdehaan) [inline content-src/components/DebugPage/DebugPage.js]: Odd, when I run this locally, I'm seeing the following error in my DevTools console:

> **TypeError: dedupedSites is not a function**

**Comment by k88hudson** [comment]
darn, looks like there's no way to disable sasslint in source code :(

File: .sass-lint.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ files:
 
 
 rules:
+  class-name-format: 0
```
Comment (k88hudson) [inline .sass-lint.yml]: not sure what to do about this

**Comment by pdehaan** [comment]
This is super useful. I had to change my config.production.yml to `DEVELOPMENT: true` locally for it to show the inspector (which makes sense).

:+1:

File: .sass-lint.yml (hunk 3 above)
Comment (pdehaan) [inline .sass-lint.yml]: Not sure. Doesn't look like **sass-lint** currently supports disabling rules per-block or per-file (see https://github.com/sasstools/sass-lint/issues/70), so we either have to disable the rule globally here, or just try and ignore the DebugPage.scss using a glob or something.

**Pull request merged**